cmd/mt-whisper-importer-reader/conversion_test.go

```diff
@@ -0,0 +1,1295 @@
+	inData := []whisper.Point{
+		{0, 0},
+		{10, 10},
+		{0, 0},
+		{0, 0},
+		{40, 13},
+		{50, 14},
+		{0, 0},
+	}
+
+	expectedResult := map[string][]whisper.Point{
+		"max": {
+			{5, 10},
+			{10, 10},
+			{35, 13},
+			{40, 13},
+			{45, 14},
+			{50, 14},
+		},
+	}
+
+	testIncResolution(t, inData, expectedResult, "max", 10, 5, 1)
+}
+
+func TestIncResolutionOutOfOrder(t *testing.T) {
+	inData := []whisper.Point{
+		{40, 13},
+		{10, 10},
+		{50, 14},
+	}
+
+	expectedResult := map[string][]whisper.Point{
+		"max": {
+			{5, 10},
+			{10, 10},
+			{35, 13},
+			{40, 13},
+			{45, 14},
+			{50, 14},
+		},
+	}
+
+	testIncResolution(t, inData, expectedResult, "max", 10, 5, 1)
+}
+
+func TestDecResolutionSimple(t *testing.T) {
+	inData := []whisper.Point{
+		{10, 10},
+		{20, 11},
+		{30, 12},
+		{40, 14},
+		{50, 15},
+		{60, 16},
+	}
+
+	expectedResult := map[string][]whisper.Point{
+		"sum": {
+			{30, 33},
+			{60, 45},
+		},
+	}
+	testDecResolution(t, inData, expectedResult, []string{"sum"}, 10, 30)
+}
+
+func TestDecResolutionUpToTime(t *testing.T) {
+	inData := []whisper.Point{
+		{10, 10},
+		{20, 11},
+		{30, 12},
+		{40, 14},
+		{50, 15},
+		{60, 16},
+	}
+
+	expectedResult := map[string][]whisper.Point{
+		"sum": {
+			{30, 33},
+		},
+	}
+	*importUpTo = uint(40)
+	testDecResolution(t, inData, expectedResult, []string{"sum"}, 10, 30)
+	*importUpTo = math.MaxUint32
+}
+
+func TestDecResolutionAvg(t *testing.T) {
+	inData := []whisper.Point{
+		{10, 10},
+		{20, 11},
+		{30, 12},
+		{40, 14},
+		{50, 15},
+		{60, 16},
+	}
+
+	expectedResult := map[string][]whisper.Point{
+		"avg": {
+			{30, 11},
+			{60, 15},
+		},
+	}
+	testDecResolution(t, inData, expectedResult, []string{"avg"}, 10, 30)
+}
+
+func TestDecNonFactorResolutions(t *testing.T) {
+	inData := []whisper.Point{
+		{10, 10},
+		{20, 11},
+		{30, 12},
+		{40, 13},
+		{50, 14},
+		{60, 15},
+	}
+
+	expectedResult := map[string][]whisper.Point{
+		"avg": {
+			{15, 10},
+			{30, 11.5},
+			{45, 13},
+			{60, 14.5},
+		},
+	}
+	testDecResolution(t, inData, expectedResult, []string{"avg"}, 10, 15)
+}
+
+func TestDecResolutionWithGaps(t *testing.T) {
+	inData := []whisper.Point{
+		{0, 0},
+		{10, 10},
+		{0, 0},
+		{0, 0},
+		{40, 13},
+		{50, 14},
+		{0, 0},
+		{70, 16},
+	}
+
+	expectedResult := map[string][]whisper.Point{
+		"avg": {
+			{20, 10},
+			{40, 13},
+			{60, 14},
+		},
+	}
+
+	testDecResolution(t, inData, expectedResult, []string{"avg"}, 10, 20)
+}
+
+func TestDecResolutionOutOfOrder(t *testing.T) {
+	inData := []whisper.Point{
+		{20, 11},
+		{50, 15},
+		{30, 12},
+		{10, 10},
+		{60, 16},
+		{40, 14},
+	}
+
+	expectedResult := map[string][]whisper.Point{
+		"avg": {
+			{30, 11},
+			{60, 15},
+		},
+	}
+	testDecResolution(t, inData, expectedResult, []string{"avg"}, 10, 30)
+}
+
+func testDecResolution(t *testing.T, inData []whisper.Point, expectedResult map[string][]whisper.Point, methods []string, inRes, outRes uint32) {
```

Comment:
why is this function placed right in the middle of all its callers, maybe move it to the top, right under testIncResolution

Reply:
I moved it right above its call sites